### ISSUE TITLE
Android: Only include the repos in places where we need them

### ISSUE THREAD

**Issue by WhatsEmo** [flutter/flutter#29700]
By default, Flutter generates the Android gradle scripts that most Flutter apps/plugins use. For the Flutter app generation, the gradle scripts put the follow at the top level gradle script:

```
allprojects {	
    repositories {	
        google()	
        jcenter()	
    }	
}
```
This is great practice for a top level gradle script that is managing many modules, but it’s a not so good practice for an one module app. The one module app should place this at the lower level app level gradle script. For the official Flutter script, having gold standard practices would prevent propagating this misunderstanding.

For the Flutter plugin generation, we have the following:
```
rootProject.allprojects {
    repositories {
        google()
        jcenter()
    }
}
```
It's forcing all apps that use this plugin to use those repositories and the plugin could only be used for that plugin. We should only be setting the repository only where it is needed.

References:
https://docs.gradle.org/current/userguide/multi_project_builds.html 
https://guides.gradle.org/creating-multi-project-builds/#configure_from_above

**Comment by escamoteur** [comment]
Is this this of interest?

**Comment by blasten** [comment]
Good point. We shouldn't be setting this configuration in a plugin. @WhatsEmo Feel free to send a PR to update this file in the plugin template file.
